src/wallet/rpcdump.cpp

```diff
@@ -74,19 +77,166 @@ std::string DecodeDumpString(const std::string &str) {
+            }
+            throw JSONRPCError(RPC_TRANSACTION_ERROR, state.GetRejectReason());
+        }
+
+        LogPrintf("SweepPrivKey(): Error: Transaction not valid\n");
+        return false;
+    }
+
+    if (pwalletMain->GetBroadcastTransactions() && !g_connman)
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
+
+    // Broadcast the transaction
+    CInv inv(MSG_TX, commit.GetHash());
+    g_connman->ForEachNode([&inv](CNode* pnode)
+    {
+        pnode->PushInventory(inv);
+    });
+
+    return true;
+}
+
+UniValue sweepkey(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() != 2)
+        throw runtime_error(
+            "sweepkey \"bitcoinprivkey\" \"bitcoinaddress\"\n"
+            "\nSweep the privkey's outputs to the specified address.\n"
+            "\nArguments:\n"
+            "1. \"bitcoinprivkey\"   (string, required) The private key (see dumpprivkey)\n"
+            "2. \"bitcoinaddress\"   (bitcoinaddress, required) The destination address\n"
+            "\nExamples:\n"
+            "\nSweep a private key\n"
+            + HelpExampleCli("sweepkey", "\"mykey\" \"myaddress\"") +
+            "\nAs a JSON-RPC call\n"
+            + HelpExampleRpc("sweepkey", "\"mykey\" \"myaddress\"")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    EnsureWalletIsUnlocked();
+
+    string strSecret = params[0].get_str();
+    string strAddress = params[1].get_str();
+
+    CBitcoinSecret vchSecret;
+    if (!vchSecret.SetString(strSecret))
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid private key encoding");
+
+    CBitcoinAddress address(strAddress);
+    if (!address.IsValid())
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid Bitcoin address");
+
+    if (!SweepPrivKey(vchSecret, address))
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "SweepPrivKey failed!");
+
+    return NullUniValue;
+}
+
 UniValue importprivkey(const UniValue& params, bool fHelp)
 {
     if (!EnsureWalletIsAvailable(fHelp))
         return NullUniValue;
-    
-    if (fHelp || params.size() < 1 || params.size() > 3)
+
+    if (fHelp || params.size() < 1 || params.size() > 5)
         throw runtime_error(
             "importprivkey \"bitcoinprivkey\" ( \"label\" rescan )\n"
             "\nAdds a private key (as returned by dumpprivkey) to your wallet.\n"
             "\nArguments:\n"
             "1. \"bitcoinprivkey\"   (string, required) The private key (see dumpprivkey)\n"
             "2. \"label\"            (string, optional, default=\"\") An optional label\n"
             "3. rescan               (boolean, optional, default=true) Rescan the wallet for transactions\n"
+            "4. sweepkey             (boolean, optional, default=false) Sweep imported key for improved security\n"
```

Comment:
@CryptAxe I think @luke-jr means that the risk is having funds to the imported key. Is this what you mean, @luke-jr ?  I'm not sure how this would happen, but as part of making a key autoswept, I would hope this pull would make it less easy to select for receipt of funds as part of this.